cmd/api/src/api/v2/agi_test.go

```diff
@@ -405,8 +407,34 @@ func TestResources_CreateAssetGroup(t *testing.T) {
 		Require().
 		ResponseStatusCode(http.StatusBadRequest)
 
+	// Empty asset group name
+	jsonBody, err := json.Marshal(v2.CreateAssetGroupRequest{Name: "", Tag: "valid_tag"})
+	require.Nil(t, err)
+
+	requestTemplate.
+		WithContext(&ctx.Context{
+			Host: &url.URL{},
+		}).
+		WithBody(jsonBody).
+		OnHandlerFunc(resources.CreateAssetGroup).
+		Require().
+		ResponseStatusCode(http.StatusBadRequest)
```

Comment:
I think you can also check that the response is the right description as well but if that ends up being to annoying with this testing framework, I'd not worry too much about it.